tests/test_parallel.py

```diff
@@ -64,3 +65,8 @@ def forward(self, *args, **kwargs):
 
     module_wraper = ModuleWrapper(model)
     assert is_module_wrapper(module_wraper)
+
+
+def test_collate():
```

Comment:
It seems like that the collate function has not been tested before. And we can add full unit tests for it.